src/Microsoft.VisualStudio.Threading/JoinableTaskDependencyGraph.cs

```diff
@@ -219,6 +220,63 @@ internal static HashSet<JoinableTask> GetDependentTasksFromCandidates(IEnumerabl
             return results;
         }
 
+        /// <summary>
+        /// Computes dependency graph to clean up all potential unreachable dependents items.
+        /// </summary>
+        /// <param name="syncTask">A thread blocking sychornizing task.</param>
+        /// <param name="allReachableNodes">Returns all reachable nodes in the connected dependency graph, if unreachable dependency is found.</param>
+        /// <returns>True if it removes any unreachable items.</returns>
+        internal static bool CleanUpPotentialUnreachableDependentItems(JoinableTask syncTask, [MaybeNullWhen(false)] out HashSet<IJoinableTaskDependent>? allReachableNodes)
```

Comment:
I'm not sure why you need `[MaybeNullWhen(false)]` here. A nullable ref annotated type such as this parameter always may be null unless you say otherwise. This attribute is for when the out parameter is a generic type and you're not allowed to annotate with `?`.
As I mentioned elsewhere, I think the attribute you want here is `[NotNullWhen(true)]` to indicate that the output parameter will not be null when you return `true` from this method.

Reply:
Yeah, that is right. I will fix that.